db/c_level_db.go

```diff
@@ -150,3 +148,23 @@ func (mBatch *cLevelDBBatch) Write() {
 		PanicCrisis(err)
 	}
 }
+
+//----------------------------------------
+// CLevelDB Iterator adaptor
+
+type cLevelDBIterator struct {
+	*levigo.Iterator
+}
+
+func (c cLevelDBIterator) Next() bool {
+	c.Next()
```

Comment:
infinite recursion? maybe `c.(*levigo.Iterator).Next()`

Reply:
nice catch!

Reply:
c.Iterator.Next() should work...

Reply:
Or perhaps let's make it explicit to
```go
type cLevelDBIterator struct {
      iter *levigo.Iterator
}

func (c cLevelDBIterator) Next() bool {
      c.iter.Next()
}
```
to avoid such problems that we previously had with https://github.com/tendermint/tmlibs/commit/0b22b27bbb08947e3875dcfe22f95b983635eb0a where the problem slipped through the cracks?